src/v/raft/append_entries_buffer.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "raft/types.h"
+
+#include <seastar/core/condition-variable.hh>
+#include <seastar/core/gate.hh>
+
+namespace raft {
+
+class consensus;
+
+class append_entries_buffer {
+public:
+    explicit append_entries_buffer(consensus&, size_t max_buffered_elements);
+
+    ss::future<append_entries_reply> enqueue(append_entries_request&& r);
+
+    void start();
```

Comment:
i see why we need it in *this* implementation, but i don't get why the design space requires you to start a background fiber and not just block. 

couldn't you simply add a sequence id and then scan for flushes, and respond to all of the requests in sequence-order? 

i don't understand the design space here. 

This looks critical enough to require an RFC

Reply:
The problem is that requests have to be executed in order they were enqueued. Each request is raft `append_entry` request that is required to be processed in order. The algorithm is really simple. it is append all entries in order without flushing (do not respond yet), flush, send responses.